lib/model/query/projects.js

```diff
@@ -94,16 +94,16 @@ ${extend|| sql`
     on field_keys."projectId"=projects.id`}
 ${(actorId == null) ? sql`` : sql`
 inner join
-  (select id, min("roleId") as "bestRole" from projects
+  (select id, array_agg(distinct verb) as verbs from projects
     inner join
-      (select "acteeId", role.id as "roleId", role.verbs from assignments
+      (select "acteeId", jsonb_array_elements_text(role.verbs) as verb from assignments
         inner join (select id, verbs from roles where verbs ?& array['project.read', 'form.list']) as role
```

Comment:
:+1: it would be helpful to have a test case reflecting this...

Reply:
Yeah, what I'm suggesting isn't really relevant for our current roles. 😅 You could write a test where you create a new role along these lines, then assign it to Dave. But my instinct is also that it's OK if just the current tests continue to pass.

Reply:
I think I have a test getting at this! (Though possibly I should have split it into two&hellip;)

```js
it('should return the correct projects with the correct verbs', testService((service) =>
  service.login('alice', (asAlice) =>
    service.login('chelsea', (asChelsea) => Promise.all([
      asChelsea.get('/v1/users/current')
        .expect(200)
        .then(({ body }) => body.id),
      asAlice.post('/v1/projects')
        .send({ name: 'Another Project' })
        .expect(200)
        .then(({ body }) => body.id)
    ])
      .then(([chelseaId, projectId]) => Promise.all([
        asAlice.post(`/v1/projects/1/assignments/viewer/${chelseaId}`)
          .expect(200),
        asAlice.post(`/v1/projects/1/assignments/app-user/${chelseaId}`)
          .expect(200),
        asAlice.post(`/v1/projects/${projectId}/assignments/app-user/${chelseaId}`)
          .expect(200)
      ]))
      .then(() => asChelsea.get('/v1/projects?forms=true')
        .expect(200)
        .then(({ body }) => {
          body.length.should.equal(1);
          const project = body[0];
          project.id.should.equal(1);
          project.verbs.should.eqlInAnyOrder([
            'project.read',     // from role(s): viewer
            'form.list',        // from role(s): viewer
            'form.read',        // from role(s): viewer, app-user
            'submission.read',  // from role(s): viewer
            'submission.list',  // from role(s): viewer
            'submission.create' // from role(s): app-user
          ]);
        }))))));
```

- This test fails on `master`, because the query there only returns verbs from a single role.
- It also fails for the latest commit of this PR (345be343478c5d5df8e3e2f3b9662e3d1357d966) because the query there only returns verbs from roles that have the verbs `project.read` and `form.list`, which the `app-user` role does not.
- It also fails for the latest commit of #535 (290d1b86df46b6400c2e1bda3794ebecc638cfb1) because the query there doesn't filter on verbs at all. /v1/projects?forms=true should only return projects for which the user can `project.read` and `form.list`, which means that it should only return a single project in this test.

I threw out this idea when we chatted today, but I think the answer is probably to remove the `WHERE` clause, as 290d1b86df46b6400c2e1bda3794ebecc638cfb1 does today, but then to also add a `HAVING` clause that does that filtering at a more outer stage, something like:

```sql
(select id, array_agg(distinct verb) as verbs from projects
  inner join -- ...
  group by id
  having array_agg(distinct verb) @> array['project.read', 'form.list']) as filtered
```

Reply:
:+1: I've added the test case and corresponding fix to the PR